packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1017,6 +1017,14 @@ abstract class State<T extends StatefulWidget> with Diagnosticable {
   /// return a future (the callback cannot be `async`), since then it would be
   /// unclear when the state was actually being set.
   ///
+  /// In fact, this callback is a simple convention that allows anyone reading
+  /// the code to easily catch the reason of why [setState] is called. Therefore,
+  /// it should be never left empty, if that's impossible, put a comment inside it:
```

Comment:
I am not sure what this is saying, can you elaborate on what you are trying to say here?

Reply:
i'm trying to explain why there's a callback there. for me when i looked up at the source code, it was opaque, since, it's just called immediately and that's it. i expressed hixie's comment https://github.com/flutter/flutter/issues/12296#issuecomment-332998974 in a form that's suitable to be in documentation

Reply:
Oh I see, thank you for the additional context! I feel maybe that means we are missing some more context here. I don't really understand that from what you have written.

Also, since the documentation says `it should be never left empty`, we should probably not provide example code of an empty `setState`.